tests/framework/installer/ceph_installer.go

```diff
@@ -680,6 +680,10 @@ func (h *CephInstaller) UninstallRookFromMultipleNS(manifests ...CephManifests)
 			continue
 		}
 
+		// remove mon finalizers
+		h.removeConfigMapFinalizers(namespace, "rook-ceph-mon-endpoints")
```

Comment:
Actually, I wonder if the finalizers on the secret and configmap will already be removed by the operator in the call to `removeFinalizers()`. Can you try executing the tests without removing the finalizers here in the test code? If they aren't removed, then it sounds like a bug.

Reply:
IMO the Finalizers needs to be removed as we are not calling the `removeFinalizers()` method or its parent form integration tests, 
and also we do similar for cephcluster CR finalizer removal.
[`func (h *CephInstaller) removeClusterFinalizers(namespace, clusterName string) {`](https://github.com/rook/rook/blob/bcae365cb9c1d87a6fc9f43d3e8471df13858ae0/tests/framework/installer/ceph_installer.go#L779)

PS: Tried without removal in second commit, it shows timeout error on upgrade test,

Reply:
Checked the logs of the failing integration tests.  I don't think its calling `[removeClusterFinalizers](https://github.com/rook/rook/blob/bcae365cb9c1d87a6fc9f43d3e8471df13858ae0/tests/framework/installer/ceph_installer.go#L779)` method in the integration tests.  So `cephCluster` finalizer  must be getting removed by the operator itself. So its strange why cephClusters finalizers are getting removed but not the new mon and configmap finalizers.

Reply:
Looks like some issue with the implementation.The [RemoveFinalizer](https://github.com/rook/rook/blob/fcf634e3b7808e4338e7fd261c047aae4793e99b/pkg/operator/ceph/controller/finalizer.go#L77) method constructs the finalizer name and then tries to remove it.  For example- In case of cephCluster, its looking for `- cephcluster.ceph.rook.io`.   The new finalizer being added (for example : `- ceph.rook.io/disaster-protection` ) does not match this format.  So it might never remove the new finalizer from the secert and configmap.

Reply:
Yes right,

`configmap.ceph.rook.io`
`secret.ceph.rook.io`
It will build the following names for the mon configmap and secret finalizers resp, 

So should we consider changing their resp names? 

or changing the [RemoveFinalizer ](https://github.com/rook/rook/blob/fcf634e3b7808e4338e7fd261c047aae4793e99b/pkg/operator/ceph/controller/finalizer.go#L77 )method(that can be done considering we would be always adding one finalize for a particular resource)

Reply:
my suggestion would be update the RemoveFinalizer method to pass in the `objectFinalizer` as an argument. This would make the method more reusable.

Reply:
Yes seems good way :)